examples/nsl_pk/DY.Example.NSL.SecurityProperties.fst

```diff
@@ -22,41 +22,46 @@ open DY.Example.NSL.Protocol.Stateful.Proof
 /// unless the attacker corrupted Alice or Bob.
 
 val initiator_authentication:
+  {|protocol_invariants|} ->
   tr:trace -> i:timestamp ->
   alice:principal -> bob:principal -> n_a:bytes -> n_b:bytes ->
   Lemma
   (requires
     event_triggered_at tr i bob (Respond2 alice bob n_a n_b) /\
-    trace_invariant tr
+    trace_invariant tr /\
+    has_nsl_invariants
```

Comment:
It feels a little bit strange in the context of these files (where it's not defined earlier in the file, as in NSL.Protocol.Stateful.Proof) that `has_nsl_invariants` takes no arguments --- it reads as a predicate on something (which, I suppose, it is, just with an implicit argument). 

Maybe there's some renaming that helps with this? `invariants_have_nsl_invariants` feels a bit strange as well, though it does describe what the predicate does. It might also be fine as is and users would just need to get used to it. It's also not core, so a user can name this kind of predicate however they want, but it would be nice if we could find a good (clear) way to present it for examples.

Reply:
Correct. Maybe something like `global_invariants_have_nsl_invariants`? Or `protocol_invariants_have_nsl_invariants`?
I proposed to switch to the typeclass-style in #47 because it's annoying when it's for `crypto_usages` or `crypto_invariants`, because then you would need to write by hand `invs.crypto_invs.usages`.
Whatever the name is, we should definitely add comments to explain this!